rest/model/patch.go

```diff
@@ -42,6 +42,7 @@ type APIPatch struct {
 	PatchedConfig           *string           `json:"patched_config"`
 	CanEnqueueToCommitQueue bool              `json:"can_enqueue_to_commit_queue"`
 	ChildPatches            []ChildPatch      `json:"child_patches"`
+	ChildPatchesTemp        []APIPatch        `json:"child_patches_temp"`
```

Comment:
do we intend to in the future rename this to "child_patches" or will that break things? In which case we should call this something more descriptive, like ChildPatchesFull

Reply:
Yup, it will be renamed and `ChildPatches` will be removed! I discussed with Chaya and the process will go roughly like this:

1. Keep `childPatches` as-is; add `childPatchesTemp` field using Patch type (this PR)
2. Make front-end changes pointing to `childPatchesTemp` (spruce)
3. Update `childPatches` such that it is identical to `childPatchesTemp` (evg)
4. Point front end to `childPatches` (spruce)
5. Remove `childPatchesTemp` (evg)

Reply:
I wonder if we can use "child_patches" here, since this will stick around, and change the json for ChildPatches to something else.

Reply:
I think the workflow Sophie described makes sense; we can't use child_patches for the new field and have it be backwards compatible since Spruce is currently using `child_patches` and expecting `[]ChildPatch` I think

Reply:
spruce is expecting []ChildPatch, but is it using the json tag 'child_patches' directly?

Reply:
Normally we would want to avoid having two different types associated with the same key in the db. However, since this in not yet used it prod, it might still be doable to try to reuse 'child_patches' to avoid calling it something that is less readable. It's not the end of the world though if we have to call it something like 'child_patches_full'.

Reply:
> spruce is expecting []ChildPatch, but is it using the json tag 'child_patches' directly?

Not as far as I can tell!

Reply:
Oh if we aren't using child_patches in spruce yet then there shouldn't be a reason to go through the backward compatibility steps.

Reply:
there are no childPatch docs in prod, but the query is already set up in spruce.

Reply:
Spruce isn't using the JSON tag directly, but it _is_ already expecting `[]ChildPatch`. This approach attempts to avoid requiring a timed release between Evergreen and Spruce in order to change `ChildPatch` into a `Patch` type. Let me know if I can clarify further since you're just jumping in here! 😊

Reply:
JK @malikchaya2 made a great point that I didn't realize:

> nobody is using childPatches yet. we can technically just comment out the query in Spruce and deploy
> 
> then we can just directly make the changes on evergreen to change ChildPatches to a Patch type

Will have another go at this, third time's the charm! 😆